backends/imgui_impl_osx.mm

```diff
@@ -630,28 +663,40 @@ bool ImGui_ImplOSX_HandleEvent(NSEvent* event, NSView* view)
         if (event.phase == NSEventPhaseCancelled)
             return false;
 
-        double wheel_dx = 0.0;
-        double wheel_dy = 0.0;
-
         #if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6)
         {
-            wheel_dx = [event scrollingDeltaX];
-            wheel_dy = [event scrollingDeltaY];
+            double wheel_dx = [event scrollingDeltaX];
+            double wheel_dy = [event scrollingDeltaY];
             if ([event hasPreciseScrollingDeltas])
             {
                 wheel_dx *= 0.1;
                 wheel_dy *= 0.1;
             }
+
+            if (event.momentumPhase)
+            {
+                // Represents a flick gesture from a track pad
+                // or magic mouse, so we want to affect the scroll position
+                // every frame to, which reproduces macOS's scroll views which
+                // use acceleration.
+                io.MouseWheelH = wheel_dx * 0.1;
+                io.MouseWheel  = wheel_dy * 0.1;
```

Comment:
Hey @stuartcarnie, i am in the process of finishing up this PR and i need some feedback. I am splitting some parts of this PR that will be applied to master branch while leaving actual viewports stuff for docking.

Starting with this change, could you explain why this branch modifies `io.MouseWheel` directly while another branch seemingly applies same input, except through events? We should not side-step event system here so i wonder what you tried to do here.

Reply:
Yes, I have done this to maintain the acceleration properties of a track pad, when scrolling. This results in the view slowly decelerating and feels more at home on macOS. Using the events is quite jarring when you are used to flicking and watching views decelerate.

Reply:
Thanks! This need to be looked at and moved to a separate issue IHMO.

Reply:
@stuartcarnie could you change these lines to also use `io.AddMouseWheelEvent((float)wheel_dx * 0.1, (float)wheel_dy * 0.1);` and then comment out `mouse_wheeled ||` bit on [this line](https://github.com/ocornut/imgui/blob/b5b704d376ae9ba2fa2e42f8728ef489d4499efd/imgui.cpp#L7843) and see if it is any better compared to using `AddMouseWheelEvent()` only?

Testing this behavior requires at very least a trackpad, right? Or can we test this with plain old PC mouse?

Reply:
It does not work, because the events added to the queue are not replayed at the same time they are received, but in batches. The trackpad NSEvent events are synchronised to the display so the acceleration is perceived in realtime.

Reply:
How about commenting `if (...) break;` entirely? That should be submitting events immediately. Could you please test?

Reply:
Another detail: when `hasPreciseScrollingDeltas` and `hasPreciseScrollingDeltas` are set, deltas would be multiplied by `0.1` twice, reducing scrolling deltas to 1% of original value. Is that intended? Seems like it should make scrolling extremely slow.